src/main/java/seedu/address/model/meeting/MeetingDate.java

```diff
@@ -23,19 +31,30 @@
     public MeetingDate(String date) {
         requireNonNull(date);
         checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
-        value = date;
+        value = LocalDate.parse(date, formats);
     }
 
     /**
-     * Returns if a given string is a valid date.
+     * checks whether the given date is valid
      */
     public static boolean isValidDate(String test) {
-        return test.matches(VALIDATION_REGEX);
+        try {
+            LocalDate.parse(test, formats.withResolverStyle(ResolverStyle.STRICT));
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+        return true;
     }
 
+    public LocalDate getValue() {
+
+        return this.value;
+    }
+
+
     @Override
     public String toString() {
-        return value;
+        return value.format(DateTimeFormatter.ofPattern("dd MMM yyyy"));
```

Comment:
This breaks the saving and loading of data in the Storage component. The dates are being saved in the JSON file as `"dd MMM yyyy"`, so when the data is loaded, the MeetingDate object cannot be created since `"dd MMM yyyy"` fails `MeetingDate::isValidDate`